### ISSUE TITLE
Notifications don’t always open on Android when tapping on a Push Notification

### ISSUE THREAD

**Issue by daniloercoli** [wordpress-mobile/WordPress-Android#4571]
### Expected behavior
When tapping on a WordPress push notification in the device notification center, the full notification content should open inside the app.

### Actual behavior
I tap on the notification and the WordPress app opens on the notifications tab showing the full list of all my read notifications, but the one that I tapped on is not in the list, as if it doesn’t actually exist. 
(The toast "Could not open the notification" can appear for few secs).

If I then wait for a while (anything from a few seconds to a minute or two), the unread notification will appear at the top. 
Sometimes I have to kill the app completely and reopen it to get the notification to be displayed in the list.

### Steps to reproduce the behavior
This seems to happen mostly when I’m on a slower internet connection (mobile data with poor reception or unreliable wifi), so what I’m guessing is happening is that the push notification is received, but the synch with Simperium fails, or it's not started at all.
Btw, sometimes it happened also when I was on fast connections.

##### Tested on Nexus 5x Android 7.0

**Comment by daniloercoli** [comment]
Just happened again. On a Wi-Fi connection (maybe a bit busy, but definitively working).
It seems that the Simperium bucket is not yet started when the app tries to load the new note, and so the new note could not be there from the db.

Push notification just arrived, and Simperium is a bit late.

I think if you wait a bit, wp-android/Simperium should synch in background, and so, later, if you open the app by tapping on the notification it should work without problem. 

WordPress Android version: alpha-21 Android device name: LGE Nexus 5X 
01 - WordPress.onCreate 
02 - Received Message 
03 - Received Message 
04 - Received Message 
05 - trackLastActivity, activityId: Notifications 
06 - App comes from background 
07 - ConnectionChangeReceiver.setEnabled true 
08 - Connection status changed, isConnected=true 
09 - Deferred Initialisation 
**10 - Configuring Simperium** 
11 - Number of old stats deleted : 0 
12 - trackLastActivity, activityId: Notifications 
13 - App Startup: begin 
14 - App Startup: 1503795 ms, WPMainActivity.onCreate 
15 - App Startup: 419 ms, WPMainActivity.onResume 
16 - App Startup: end, 1504214 ms 
17 - Creating NotificationsDetailActivity 
18 - Sending GCM token to our remote services: fFBYrA9B
**19 - Note could not be found.** 
20 - trackLastActivity, activityId: Notifications 
21 - reader post list > newInstance (tag) 
**22 - Starting Simperium buckets**
23 - reader post list > purging database 
24 - reader post list > updating tags and blogs 
25 - reader post list > creating post adapter 
26 - reader service > created 
27 - reader service > updating tags 
28 - reader service > updating followed blogs 
29 - reader post list > updating tag a8c, updateAction=REQUEST_NEWER 
30 - Register token action succeeded 
31 - Server response OK. The device_id: 15230457 
32 - reader post service > created 
33 - requested posts response = HAS_NEW 
34 - reader post service > destroyed 
35 - Received Message 
36 - Received Message 
37 - reader post list > saving instance state 
38 - reader post list > restoring instance state 
39 - trackLastActivity, activityId: Notifications

**Comment by mzorz** [comment]
After considering this for a while, I came up with 3 alternative solutions:

a) we just show a  ‘We couldn’t load the note, sorry”
b) we show a progress sign so the user waits until we sync simperium (see if we have callbacks for this)
c) we try fetching the note detail from the REST api, but for this we need to know the remote blog id and local blog id, which we should be sending along with the PN payload (we aren’t doing so right now).

The first one is self explanative, and the easiest one to solve (but provides a poor user experience, better though than just showing a blank Comment screen).

The second one only requires changes on the client side, but we need to know when the sync has completed.

The third one while probably is the most complete in terms of the best effort we can do to show the user what the user is supposed to see, it requires changes on both Android and the server, and also it pretty much means replicating the functionality that Simperium already is supposed to be taking care of (aka for comments we would need to send the remote site id & comment id in the PN payload and then be able to fetch it from the REST api, and we’d also need to send some other params for each different type of note as needed). 

After trying some things I decided the third case is the best, so I am working on this third solution now (which also, in case it fails to load anything at all, falls back to solution (a) ). For now I’m only focusing in writing the solution for Comments only, which seems the particular place where most of these cases have appeared. Will post a PR shortly. 👍
